cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -2360,6 +2361,39 @@ TEST(ArrowReadWrite, SingleColumnNullableStruct) {
       3);
 }
 
+TEST(ArrowReadWrite, NestedRequiredField) {
```

Comment:
The test cases look very very similar, just the name and the used values differ. I would have expected that we also would have set `nullable=false` somewhere in this one.

Reply:
Are you looking at the latest version?

Right below this comment is:
```
auto int_field = ::arrow::field("int_array", ::arrow::int32(), /*nullable=*/false);
```
(note the last parameter?)

Reply:
Ah, didn't see that when reviewing this code. Now this makes sense!